src/assignment/views/AssignmentEdit.tsx

```diff
@@ -238,34 +237,15 @@ const AssignmentEdit: FunctionComponent<AssignmentEditProps> = ({
 				return;
 			}
 
-			if (tempAssignment.contributors && user && user?.profile?.id) {
-				const contributorInfo = tempAssignment.contributors.find(
-					(contributor) => contributor.profile_id === user?.profile?.id
-				);
-				const isOwner = tempAssignment.owner_profile_id === user.profile.id;
-
-				if (
-					(!contributorInfo && !isOwner && !permissions.canEditAllAssignments) ||
-					(contributorInfo?.rights === Lookup_Enum_Right_Types_Enum.Viewer &&
-						!permissions.canEditAllAssignments)
-				) {
-					setAssignmentError({
-						message: tHtml(
-							'assignment/views/assignment-edit___je-hebt-geen-rechten-om-deze-opdracht-te-bewerken'
-						),
-						icon: IconName.lock,
-						actionButtons: ['home'],
-					});
-					setAssigmentLoading(false);
-					return;
-				}
-			}
-
 			const hasPupilBlocks = await AssignmentService.hasPupilCollectionBlocks(id);
 
 			const checkedPermissions = await PermissionService.checkPermissions(
 				{
-					canEditAllAssignments: [
+					canEdit: [
+						{
+							name: PermissionName.EDIT_OWN_ASSIGNMENTS,
+							obj: assignmentId,
+						},
```

Comment:
this adds back an additional call to the backend to fetch the assignment
but we already fetched it, and the backend checked that we can view the assignment

so the old code was more efficient